packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -963,6 +980,9 @@ export class IndexedDbPersistence implements Persistence {
    * handler.
    */
   private attachWindowUnloadHook(): void {
+    if (!this.window) {
+      return;
+    }
     if (typeof this.window.addEventListener === 'function') {
```

Comment:
Not sure what effect this has on code size, but you could remove some of extra lines in this PR but using optional chaining:

```
if (typeof this.window?.addEventListener === 'function') {}
this.webStorage?.setValue(...)
```